RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -279,6 +294,7 @@ void PhotonProducer::fillPhotonCollection(edm::Event& evt,
 					  const CaloTopology* topology,
 					  const EcalRecHitCollection* ecalBarrelHits,
 					  const EcalRecHitCollection* ecalEndcapHits,
+					  const EcalRecHitCollection* preshowerHits,
```

Comment:
What are the remaining differences in ::fillPhotonCollection  compared to the GEDPhotonProducer after the updates proposed in this PR?
It looks like the right solution is to refactor and use the same method in both cases.